src/Extensions/Abstractions/ExecutionContext/BaseExecutionContext.cs

```diff
@@ -137,6 +143,158 @@ protected static string GetBuildVersion()
 				: DefaultEmptyValue;
 		}
 
+		/// <summary>
+		/// Get build version from the current running service's manifest file
+		/// </summary>
+		/// <param name="serviceManifestName"> Service manifest name defined in ApplicationManifest.xml </param>
+		/// <returns> Build version if found, otherwise null </returns>
+		protected static string? GetBuildVersionFromServiceManifest(string serviceManifestName)
+		{
+			string? serviceManifestPath = GetServiceManifestPath(serviceManifestName);
```

Comment:
Is this the only way to achieve it? Is it maybe be possible to use some of the SF libraries?